lib/rules/class-methods-use-this.js

```diff
@@ -103,6 +114,20 @@ module.exports = {
             }
         }
 
+        /**
+         * Check if the node's parent class implements any interfaces
+         * @param {ASTNode} node node to check
+         * @returns {boolean} True if parent class implements interfaces
+         * @private
+         */
+        function hasImplements(node) {
+            const classNode = node.parent.parent;
+            return classNode && 
+                   classNode.type === "ClassDeclaration" && 
+                   classNode.implements && 
+                   classNode.implements.length > 0;
```

Comment:
We can shorten this a bit
```suggestion
            return classNode?.type === "ClassDeclaration" &&
                   classNode.implements?.length > 0;
```